src/Nethermind/Nethermind.Blockchain/Producers/TxPoolTxSource.cs

```diff
@@ -122,7 +122,7 @@ bool HasEnoughFounds(IDictionary<Address, UInt256> balances, Transaction transac
             {
                 if (gasRemaining < Transaction.BaseTxGasCost)
                 {
-                    continue;
+                    break;
```

Comment:
If we don't have room for BaseTxGasCost there is no point iterating over this collection, which is lazy and can take potentially long time.